pkg/index/index.go

```diff
@@ -31,7 +30,14 @@ func NewTreeIndex() Index {
 	}
 }
 
-func (ti *treeIndex) Put(key []byte, rev Revision) {
+// Put inserts a new index entry
+// for now, assuming no error yield within this method
```

Comment:
There is a panic instead of error to follow fail early patterns.

Reply:
this PR does not try to change the error handling behavior yet. will update comment to avoid misunderstanding